src/ray/rpc/grpc_server.cc

```diff
@@ -48,7 +47,15 @@ void GrpcServer::PollEventsFromCompletionQueue() {
         // incoming request.
         server_call->GetFactory().CreateCall();
         server_call->SetState(ServerCallState::PROCESSING);
-        main_service_.post([server_call] { server_call->HandleRequest(); });
+        if (!main_service_.stopped()) {
+          main_service_.post([server_call] { server_call->HandleRequest(); });
+        } else {
+          // Handle service for rpc call has stopped, we must handle the call here
+          // to send reply and remove it from cq
+          RAY_LOG(DEBUG) << "Handle service has been closed.";
+          server_call->Finish(Status::Invalid("HandleServiceClosed"));
+          delete_call = true;
```

Comment:
should we delete this call here? 
I think when the reply is sent, another event will be triggered. we should delete this call at that time.

Reply:
yes, double free may occur here!